tests/unit/configFile.spec.js

```diff
@@ -133,4 +133,239 @@ describe("configFile.js", () => {
       expect(result).toStrictEqual(changedSettings);
     });
   });
+
+  describe("parseBasicSettings function", () => {
+    it("data should be parsed correctly", async () => {
+      await LocalStore.clear();
+      const expectedParsedData = {
+        configurationURL:
+          "https://raw.githubusercontent.com/AdvancedThreatAnalytics/threat-analytics-search/main/settings.json",
+        useGroups: true,
+        configEncrypted: false,
+        configEncryptionKey: "",
+        autoUpdateConfig: false,
+      };
+      const result = await ConfigFile.default.parseBasicSettings(
+        defaultSettings.config
+      );
+
+      expect(result).toEqual(expect.objectContaining(expectedParsedData));
+      expect(result.mergeGroups).toEqual(result.useGroups);
+    });
+  });
+
+  describe("parseGroups function", () => {
+    it("only array of length 3 are parsed, result should be an array of object with first two being enabled", async () => {
+      const groups = [
+        ["1", "IP Lookup"],
+        ["2", "Domain"],
+        ["3", "Hash"],
+      ];
+      const expectedResult = [
+        { name: "IP Lookup", enabled: true },
+        { name: "Domain", enabled: true },
+        { name: "Hash", enabled: false },
+      ];
+      var result = await ConfigFile.default.parseGroups(groups.slice(0, 1));
+      expect(result).toHaveLength(0);
+
+      result = await ConfigFile.default.parseGroups(groups);
+      expect(result).toHaveLength(3);
+      expect(result).toStrictEqual(expectedResult);
+    });
+  });
+
+  describe("updateSpecialProvider function", () => {
+    it("configuration is overridden only if the corresponding flag is enabled", async () => {
+      const newRSA = _.cloneDeep(defaultSettings.RSA);
+      newRSA.Config.RSAConfigEnable = true;
+
+      // Configuration shouldn't be changed if flag is false.
+      await ConfigFile.default.updateSpecialProvider(
+        StoreKey.RSA_SECURITY,
+        newRSA,
+        "mergeRSA"
+      );
+      var result = await LocalStore.getOne(StoreKey.RSA_SECURITY);
+      expect(result.config.RSAConfigEnable).toEqual(
+        defaultSettings.RSA.Config.RSAConfigEnable
+      );
```

Comment:
Here you are testing that the `RSAConfigEnable` flag hasn't changed, but you should compare several fields (not necessarily all of them, but at least three or four).

> Also, if later we modify `settings.json` so `RSAConfigEnable` is defined as true, then this test will be invalid (because both the new and old value will be true, so the test will always succeed, despite the data being overridden or not).